clo-workflows/Scripts/src/store/ClientStore.ts

```diff
@@ -180,14 +137,140 @@ export class ClientStore {
             accumulator[fieldName] = error
             return accumulator
         }, {})
-        //   }
-        //   @computed get selectedProcessValidation(): {} {
-        //     return this.selectedProcessFormControls.reduce((accumulator: {}, formControl: IFormControl) => {
-        //         const fieldName: string = formControl.dataRef
-        //         const inputVal = this.selectedProcess.get(fieldName)
-        //         const error: string = inputVal ? validateFormControl(formControl, inputVal) : null
-        //         accumulator[fieldName] = error
-        //         return accumulator
-        //     }, {})
+    }
+    /************ Actions ***************/
+    @action
+    async updateObject(fieldName: string, newVal: FormEntryType, objToUpdate?: OBJECT_TYPES) {
```

Comment:
this abstraction is a little confusing to me. I would say either have explicit actions called updateWork, updateProject, etc... or rename the object_types enum to something a little more descriptive. Also it looks this function is doing two distinct things: this[objToUpdate].set(fieldName, newVal) and this[fieldName] = newVal. Its a little cryptic